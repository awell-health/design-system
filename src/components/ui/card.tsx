import * as React from "react"

import { cn } from "@/lib/utils"

const Card = React.forwardRef<
  HTMLDivElement,
  React.HTMLAttributes<HTMLDivElement>
>(({ className, children, ...props }, ref) => (
  <div
    ref={ref}
    className={cn(
      "rounded-xl border shadow p-3",
      className
    )}
    {...props}
  >{children}</div>
))
Card.displayName = "Card"



export { Card }
