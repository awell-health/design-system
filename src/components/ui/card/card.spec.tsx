import { render } from '@testing-library/react';
import { expect, it, describe } from 'vitest';
import { Card } from './card';

describe('Card', () => {
  it('match snapshot', () => {
    const { container } = render(
      <Card>
        <div>Card Content</div>
      </Card>
    );

    expect(container).toMatchSnapshot();
  });
});
