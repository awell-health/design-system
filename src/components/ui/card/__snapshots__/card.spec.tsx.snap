// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Card > match snapshot 1`] = `
<div>
  <div
    class="rounded-xl border shadow p-3"
  >
    <div>
      Card Content
    </div>
  </div>
</div>
`;
